receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -490,7 +491,19 @@ func parseMessageToMetric(line string, enableMetricType bool, enableSimpleTags b
 		result.description.attrs = attribute.NewSet(kvs...)
 	}
 
-	return result, nil
+	valueStrParts := strings.Split(valueStr, ":")
+	results := make([]statsDMetric, len(valueStrParts))
+	for i, valueStrPart := range valueStrParts {
+		var err error
+		result.asFloat, err = strconv.ParseFloat(valueStrPart, 64)
+		if err != nil {
+			return nil, fmt.Errorf("parse metric value string: %s", valueStr)
+		}
+
+		results[i] = result
+	}
+
+	return results, nil
```

Comment:
thinking about this more - I think going from `statsDMetric` to `[]statsDMetric` is completely unnecessary if we incrementally decode the floats, eg.
```
	var valueStrPart string
	for len(valueStr) > 0 {
		if idx := strings.IndexByte(valueStr, ':'); idx == -1 {
			valueStrPart = valueStr
			valueStr = nil
		} else {
		 	valueStrPart = valueStr[:idx]
			valueStr = valueStr[idx+1:]
		}
		var err error
		result.asFloat, err = strconv.ParseFloat(valueStrPart, 64)
		if err != nil {
			return nil, fmt.Errorf("parse metric value string: %s", valueStr)
		}
		// do something with result ..
	}
```